compiler/x/codegen/X86Ops.hpp

```diff
@@ -1388,6 +1388,67 @@ class TR_X86OpCode
 
    uint8_t getOpCodeLength() {return _binaryEncodings[_opCode]._length;}
 
+   uint8_t length(uint8_t rex = 0)
+      {
+      uint8_t len = 0;
+      if (needsSSE42OpcodePrefix() || needs16BitOperandPrefix())
+         {
+         len++; // 66
+         }
+      if (needsRepPrefix() || needsXreleasePrefix() || needsXacquirePrefix() || needsScalarPrefix())
+         {
+         len++; // f3/f2
+         }
+      if (needs64BitOperandPrefix())
+         {
+         rex |= 0x48;
+         }
+      if (rex)
+         {
+         len++;
+         }
+      if (needsSSE42OpcodePrefix())
+         {
+         len++; // 0f
+         }
+      return len + getOpCodeLength();
+      }
+   uint8_t* binary(uint8_t* cur, uint8_t rex = 0)
+      {
+      uint8_t* start = cur;
+      if (needsSSE42OpcodePrefix() || needs16BitOperandPrefix())
+         {
+         *cur++ = 0x66;
+         }
+      if (needsRepPrefix() || needsXreleasePrefix())
+         {
+         *cur++ = 0xf3;
+         }
+      else if (needsXacquirePrefix())
+         {
+         *cur++ = 0xf2;
+         }
+      else if (needsScalarPrefix())
+         {
+         *cur++ = singleFPOp() ? 0xf3 : 0xf2;
+         }
+      if (needs64BitOperandPrefix())
+         {
+         rex |= 0x48;
+         }
+      if (rex)
+         {
+         *cur++ = rex;
+         }
+      if (needsSSE42OpcodePrefix())
+         {
+         *cur++ = 0x0f;
+         }
+      cur = copyBinaryToBuffer(cur);
+      TR_ASSERT(cur-start == length(rex), "Actual binary length should be the same as calculated one");
```

Comment:
It makes more sense to include getOpCodeLength in the length function; length() returns the length of the whole opcode, which includes both prefixes and the actual opcode. In addition, since current binary encoding scheme arbitrarily divide the instruction into prefixes (in op properties) and opcode, only the total length has a consistent and valid meaning.